### PR TITLE
Log `basic_RVs` sampled in  `sample_*_predictive` functions

### LINKED ISSUE #5973: Log name of RVs that are being resampled in posterior predictive
Right now, the behavior of `sample_posterior_predictive` is very opaque as to which RVs are being resampled. Any RVs that are downstream of Shared variables or other variables mentioned in `var_names`, as well as variables missing in the trace will be resampled. We should output their names, similarly to how we show what variables are being sampled in `pm.sample`

For instance, in the example below it might unclear that `y` is being resampled, as it does not show in any log messages or in the output of `posterior_predictive`:

```python
import pymc as pm
with pm.Model() as m:
    x = pm.MutableData("x", 0)
    y = pm.Normal("y", x)
    z = pm.Normal("z", y, observed=0)
    idata = pm.sample(chains=1, tune=0, draws=5, random_seed=1)
    pm.set_data({"x": 100})
    pm.sample_posterior_predictive(idata, extend_inferencedata=True, random_seed=1)
    
print(idata.posterior_predictive.data_vars)  # z (chain, draw) float64 101.8 101.5 ...
print(idata.posterior_predictive["z"].mean().values)  # 100.30893015576387
```

### PR DESCRIPTION
**Pull request by ricardoV94**
Closes #5973

## Major / Breaking Changes
- ...

## Bugfixes / New features
- ...

## Docs / Maintenance
- Log name of `basic_RVs` sampled in prior and posterior predictive

### CODE REVIEW

**Comment by ricardoV94** [comment]
Only flaky test failing: #6091

File: pymc/tests/test_sampling.py [hunk 1]
```diff
@@ -1023,6 +1024,110 @@ def test_deterministics_out_of_idata(self, multitrace):
...
+        caplog.clear()
+
+        # Missing deterministic `x_det` does not cause recomputation of downstream `y` RV
+        idata = az_from_dict(posterior={"x": np.zeros(5), "y": np.ones(5)})
+        with m:
+            pm.sample_posterior_predictive(idata)
+        assert caplog.record_tuples == [("pymc", logging.INFO, "Sampling: [z]")]
+        caplog.clear()
+
+        # Missing `x` cousse downstream of `y` RV, even if it is present in trace
```
Comment (lucianopaz) [inline pymc/tests/test_sampling.py]: Not sure what word you wanted to put here instead of cousse

**Review by lucianopaz — APPROVED** [review]
Looks great @ricardoV94 ! There was only a typo in a comment, and I’m not sure which word you wanted to put

File: pymc/tests/test_sampling.py (hunk 1 above)
Comment (ricardoV94) [reply pymc/tests/test_sampling.py]: `cousse` is a French word for I had an aneurysm.

Comment (lucianopaz) [reply pymc/tests/test_sampling.py]: Oh la la!

Comment (ricardoV94) [reply pymc/tests/test_sampling.py]: I meant `causes sampling of downstream RV y` :P

Comment (lucianopaz) [reply pymc/tests/test_sampling.py]: https://youtu.be/JhbJnlIvfyc

Comment (lucianopaz) [reply pymc/tests/test_sampling.py]: I couldn’t resist the Simpson moment of this thread

Comment (ricardoV94) [reply pymc/tests/test_sampling.py]: That's a great sketch. Didn't know it  xD

**Comment by ricardoV94** [comment]
Merging, flaky test is documented in #6143

**Pull request merged**